tcms/testruns/views.py

```diff
@@ -38,94 +38,96 @@
 User = get_user_model()  # pylint: disable=invalid-name
 
 
-@require_POST
-@permission_required('testruns.add_testrun')
-def new(request):
-    """Display the create test run page."""
-
-    # If from_plan does not exist will redirect to plans for select a plan
-    if not request.POST.get('from_plan'):
-        messages.add_message(request,
-                             messages.ERROR,
-                             _('Creating a TestRun requires a TestPlan, select one'))
-        return HttpResponseRedirect(reverse('plans-search'))
-
-    plan_id = request.POST.get('from_plan')
-    # case is required by a test run
-    # NOTE: currently this is handled in JavaScript but in the TestRun creation
-    # form cases can be deleted
-    if not request.POST.get('case'):
-        messages.add_message(request,
-                             messages.ERROR,
-                             _('Creating a TestRun requires at least one TestCase'))
-        return HttpResponseRedirect(reverse('test_plan_url_short', args=[plan_id]))
-
-    # Ready to write cases to test plan
-    test_cases = get_selected_testcases(request)
-    test_plan = TestPlan.objects.get(plan_id=plan_id)
-
-    # note: ordered by case_id for test_show_create_new_run_page()
-    tcs_values = test_cases.select_related('author',
-                                           'case_status',
-                                           'category',
-                                           'priority').order_by('case_id')
+@method_decorator(permission_required('testruns.add_testrun'), name='dispatch')
```

Comment:
Wrt permission tests: from what I can see we have tests which include the permission label (these need to be more granual - include only that permission label and nothing else). We're missing tests for negative scenario where the user doesn't have permission assigned.